primitive-types/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2015-2017 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Primitive types shared by Substrate and Parity Ethereum.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "impl-serde", not(feature = "std")))]
```

Comment:
Why did you prefer this to `serde = ["std", "impl-serde"]`?

When trying `serde = ["std", "impl-serde"]` locally with `cargo check --no-default-features --features=serde` it seems to build fine?